### PR TITLE
Nim add operators

### LINKED ISSUE #304: Add Nim Logic/Math Operators
# Description

Logic/Math Operators for Nim aren't presently on Code Thesaurus. This issue is to add that concept to the system.

# Requirements

* Copy `web/thesauruses/_meta/operators.json` to the right Nim directory in `web/thesauruses/`
* Edit the operators.json file to accurately reflect the latest major version of Nim
* Delete the `"name":` fields before your PR
* Add in `"comment":` if you wish to explain something in the code
* Add in `"not-implemented": "true"` if the concept doesn't exist in the language. You can then leave `"code":` blank.

# Claiming This Issue

Please leave a comment below so I can assign you to it. This is to prevent multiple people from working on the same issue. 

Check out [CONTRIBUTING.md](https://github.com/codethesaurus/codethesaur.us/blob/main/CONTRIBUTING.md) for more information on how to contribute. 

Check out [https://docs.codethesaur.us/thesaurus/](https://docs.codethesaur.us/thesaurus/) for information on how Code Thesaurus works and how to add to the thesauruses.

# Need Help?

Leave a comment on here with your question, or feel free to ask [@geekygirlsarah](https://twitter.com/geekygirlsarah) on Twitter. You can also email coreteam@codethesaur.us too.

# Thanks!

We really value and appreciate the additions people make to Code Thesaurus, no matter how large or small. So thank you for taking the time to work on this!

### PR DESCRIPTION
**Pull request by juancarlospaco**
## What GitHub issue does this PR apply to?

Resolves https://github.com/codethesaurus/codethesaur.us/issues/304

## What changed and why?

- Add Nim operators.

## (If editing Django app) Please add screenshots

No

## Any additional comments or things to be aware of while reviewing?
 
No

## Reviewer Steps

- [x] Review commits
- [x] (If needed) Approve first run of workflow
- [ ] (If needed) Activate Heroku Review app
- [ ] Verify any changes in Deploy Preview
- [ ] Approve and/or merge changes

### CODE REVIEW

File: web/thesauruses/nim/operators.json [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+      "code": "isnot"
+    },
+    "and": {
+      "name": "Logical AND operator",
+      "code": "and"
+    },
+    "and_assignment": {
+      "name": "Logical AND assignment operator",
+      "code": "",
+      "not-implemented": "true"
```
Comment (geekygirlsarah) [inline web/thesauruses/nim/operators.json:155]: If a concept is `not-implemented`, you can remove the `code` line. (same for the other lines below.)

File: web/thesauruses/nim/operators.json [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+    },
+    "xor_assignment": {
+      "name": "Logical XOR assignment operator",
+      "code": "",
+      "not-implemented": "true"
+    },
+    "xnor": {
+      "name": "Logical XNOR operator",
+      "code": ""
```
Comment (geekygirlsarah) [inline web/thesauruses/nim/operators.json]: Does this exist or is it `not-implemented` too? I'm guessing maybe it's not implemented but you forgot the line?

**Comment by geekygirlsarah** [comment]
If you make the couple of minor changes, I'll merge it in!

**Comment by juancarlospaco** [comment]
What change ?.

**Comment by geekygirlsarah** [comment]
Hmm... I made review comments. (See above.)  Maybe Github doesn't email those?

**Comment by juancarlospaco** [comment]
No review comments for me.

**Review by geekygirlsarah — CHANGES_REQUESTED** [review]

**Comment by juancarlospaco** [comment]
I see it now! 🙂

**Comment by geekygirlsarah** [comment]
Oooooh, apparently I missed a step. I thought "pending" meant "they hadn't worked on it yet" not "you haven't sent it to them yet"...

image

**Comment by juancarlospaco** [comment]
Good to know, so is not visible for other people until you confirm the review, interestin'

File: web/thesauruses/nim/operators.json [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+      "right_shift_assignment"
+    ],
+    "Conditional Operators": [
+      "ternary",
+      "null_forgiving"
+    ]
+  },
+  "operators": {
+    "addition": {
+      "name": "Addition operator",
```
Comment (geekygirlsarah) [inline web/thesauruses/nim/operators.json]: Sorry, I didn't end up catching this last time. Can you remove all the `name` lines? those are pulled in from the meta language and aren't needed in this file.

(Then I'll merge it. I swear!)

**Review by geekygirlsarah — CHANGES_REQUESTED** [review]

File: web/thesauruses/nim/operators.json (hunk 3 above)
Comment (juancarlospaco) [reply web/thesauruses/nim/operators.json]: No drama. 🙂

File: web/thesauruses/nim/operators.json (hunk 2 above)
Comment (geekygirlsarah) [reply web/thesauruses/nim/operators.json]: Looks like you still have an empty code line here. Is it missing the code, or does it need to be `not-implemented`?

**Review by geekygirlsarah — CHANGES_REQUESTED** [review]

File: web/thesauruses/nim/operators.json (hunk 2 above)
Comment (juancarlospaco) [reply web/thesauruses/nim/operators.json]: Fixed.

Comment (geekygirlsarah) [reply web/thesauruses/nim/operators.json]: I'm still seeing the empty code block. I just looked it up and Nim doesn't have an `xnor`, so this should have `code` removed and `"not-implemented": true` in its place.

Comment (juancarlospaco) [reply web/thesauruses/nim/operators.json]: Sorry, you are right, should be Ok now.
The JSON changed since last time I edited it, but is a good change.

**Comment by geekygirlsarah** [comment]
Thanks for fixing that last part! I'll merge this in! 

And sign up for Hacktoberfest if you haven't! This counts towards the swag!

**Pull request merged**